src/cli/download_tracker.rs

```diff
@@ -1,291 +1,80 @@
-use std::collections::VecDeque;
-use std::fmt;
-use std::io::Write;
-use std::time::{Duration, Instant};
+use indicatif::{MultiProgress, ProgressBar, ProgressDrawTarget, ProgressStyle};
+use std::time::Duration;
 
 use crate::dist::Notification as In;
 use crate::notifications::Notification;
-use crate::process::{Process, terminalsource};
+use crate::process::Process;
 use crate::utils::Notification as Un;
-use crate::utils::units::{Size, Unit, UnitMode};
-
-/// Keep track of this many past download amounts
-const DOWNLOAD_TRACK_COUNT: usize = 5;
 
 /// Tracks download progress and displays information about it to a terminal.
 ///
 /// *not* safe for tracking concurrent downloads yet - it is basically undefined
 /// what will happen.
 pub(crate) struct DownloadTracker {
-    /// Content-Length of the to-be downloaded object.
-    content_len: Option<usize>,
-    /// Total data downloaded in bytes.
-    total_downloaded: usize,
-    /// Data downloaded this second.
-    downloaded_this_sec: usize,
-    /// Keeps track of amount of data downloaded every last few secs.
-    /// Used for averaging the download speed. NB: This does not necessarily
-    /// represent adjacent seconds; thus it may not show the average at all.
-    downloaded_last_few_secs: VecDeque<usize>,
-    /// Time stamp of the last second
-    last_sec: Option<Instant>,
-    /// Time stamp of the start of the download
-    start_sec: Option<Instant>,
-    term: terminalsource::ColorableTerminal,
-    /// Whether we displayed progress for the download or not.
-    ///
-    /// If the download is quick enough, we don't have time to
-    /// display the progress info.
-    /// In that case, we do not want to do some cleanup stuff we normally do.
-    ///
-    /// If we have displayed progress, this is the number of characters we
-    /// rendered, so we can erase it cleanly.
-    displayed_charcount: Option<usize>,
-    /// What units to show progress in
-    units: Vec<Unit>,
-    /// Whether we display progress
-    display_progress: bool,
-    stdout_is_a_tty: bool,
+    /// MultiProgress bar for the downloads.
+    multi_progress_bars: MultiProgress,
+    /// ProgressBar for the current download.
+    progress_bar: ProgressBar,
 }
 
 impl DownloadTracker {
     /// Creates a new DownloadTracker.
     pub(crate) fn new_with_display_progress(display_progress: bool, process: &Process) -> Self {
+        let multi_progress_bars = MultiProgress::with_draw_target(process.progress_draw_target());
+        if !display_progress {
```

Comment:
Nit: When `display_progress` is `false` we would like to prevent unnecessary calculation. How about the following instead (unformatted, don't commit directly)?


```suggestion
        let multi_progress_bars = MultiProgress::with_draw_target(if display_progress { process.progress_draw_target() } else { ProgressDrawTarget::hidden() });
```

Reply:
Yes, this is definitely cleaner. Thank you! :)